toolchain/driver/driver.cpp

```diff
@@ -694,7 +739,9 @@ auto Driver::Compile(const CompileOptions& options) -> RunResult {
     // Shared values will always be printed after per-file printing.
     if (options.dump_shared_values) {
       for (const auto& unit : units) {
-        unit->PrintSharedValues();
+        if (unit->IncludeInDumps()) {
+          unit->PrintSharedValues();
+        }
```

Comment:
Rather than making IncludeInDumps public, maybe PrintSharedValues should just query IncludeInDumps? I mean, it's really just for dumping, not generic past that.

Reply:
Done. Factored out all the per-CU finalization code to make this consistent with the other things we do with CUs.